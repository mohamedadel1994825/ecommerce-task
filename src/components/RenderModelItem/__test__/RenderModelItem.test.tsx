import { Model } from "@interfaces";
import { cleanup, render } from "@testing-library/react-native";
import { default as React } from "react";
import { RenderModelItem } from "../index";

describe("RenderModelItem", () => {
  afterEach(cleanup);

  const model: Model = {
    id: 1,
    name: "Test Model",
    image: "test-image-url",
    availableProducts: 5,
    quantity: 10,
    space: "test-space",
    color: "test-color",
  };

  it("should match snapshot", () => {
    const { toJSON } = render(<RenderModelItem item={model} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("should render ModelCard component with correct props", () => {
    const { getByTestId } = render(<RenderModelItem item={model} />);
    const modelCard = getByTestId("model-card");
    expect(modelCard).toBeTruthy();
  });
});
