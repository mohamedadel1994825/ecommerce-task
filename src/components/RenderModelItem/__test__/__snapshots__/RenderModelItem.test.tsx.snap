// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderModelItem should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#e9eff8",
      "borderRadius": 10,
      "height": 150,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginHorizontal": 4,
      "opacity": 1,
      "width": 113,
    }
  }
  testID="model-card"
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "height": 45,
          "marginBottom": 5,
          "width": 45,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="contain"
      source={
        {
          "uri": "test-image-url",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="model-image"
    />
  </View>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 12,
        "fontWeight": "400",
        "marginBottom": 5,
      }
    }
  >
    Test Model
  </Text>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 10,
        "fontWeight": "400",
        "marginBottom": 2,
      }
    }
  >
    test-space
     | 
    test-color
  </Text>
  <Text
    style={
      {
        "fontSize": 10,
        "marginBottom": 2,
      }
    }
  >
    5
  </Text>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 12,
        "fontWeight": "400",
      }
    }
  >
    10
  </Text>
</View>
`;
